service/compute/ext_spark_version.go

```diff
@@ -76,7 +76,28 @@ func (sv GetSparkVersionsResponse) Select(req SparkVersionRequest) (string, erro
 		if req.Latest {
 			sort.Sort(sparkVersionsType(versions))
 		} else {
-			return "", fmt.Errorf("spark versions query returned multiple results %#v. Please change your search criteria and try again", versions)
+			sort.Strings(versions)
+			// We need to check for uniqueness of the versions without the scala suffix
+			// This is because we can have multiple instances of the same DBR version but different scala versions
+			uniqueVersions := make([]string, 0, len(versions))
+			for _, version := range versions {
+				// Strip -scala2.12 and -scala2.13 from the version string for uniqueness check
+				v := strings.ReplaceAll(version, "-scala2.12", "")
+				v = strings.ReplaceAll(v, "-scala2.13", "")
+				found := false
+				for _, existing := range uniqueVersions {
+					if existing == v {
+						found = true
+						break
+					}
+				}
+				if !found {
+					uniqueVersions = append(uniqueVersions, v)
+				}
+			}
+			if len(uniqueVersions) > 1 {
+				return "", fmt.Errorf("spark versions query returned multiple results %#v. Please change your search criteria and try again", uniqueVersions)
+			}
```

Comment:
Does this mean the list size should be 2? i.e. same version with 2.12 and 2.13? If yes, we can just check these 2 values rather than sort and loop.

Reply:
@alexott, I meant that if the list size >=3, then we already error because the duplicate will only be possible for 2 different scala versions right? And if list size == 2, then we check those 2 elements

Reply:
Yes, you're right... Let me reimplement it a bit differently